### PR TITLE
Add 26 Missing GitLab Settings

### PR DESCRIPTION
**Pull request by Jitsusama**
I have discovered ~ 60 settings that have been added to GitLab's API but are currently missing in this project. In this PR I'm taking the next stab at correcting this by adding in 26 missing settings, leaving 27 left.

These settings were found by perusing documentation and code in response to [gitlab-org/terraform-provider-gitlab#6332](https://gitlab.com/gitlab-org/terraform-provider-gitlab/-/issues/6332).

### CODE REVIEW

File: types.go [hunk 1]
```diff
@@ -230,6 +230,17 @@ func Availability(v AvailabilityValue) *AvailabilityValue {
...
+// BranchProtectionDefaults represents default Git protected branch permissions.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/groups.html#options-for-default_branch_protection_defaults
+type BranchProtectionDefaults struct {
+	AllowedToPush           []int `json:"allowed_to_push,omitempty"`
+	AllowForcePush          bool  `json:"allow_force_push,omitempty"`
+	AllowedToMerge          []int `json:"allowed_to_merge,omitempty"`
+	DeveloperCanInitialPush bool  `json:"developer_can_initial_push,omitempty"`
+}
```
Comment (svanharmelen) [inline types.go:242]: I prefer to have this struct defined below the `Settings` struct.

File: settings.go [hunk 2]
```diff
@@ -462,359 +463,360 @@ func (s *SettingsService) GetSettings(options ...RequestOptionFunc) (*Settings,
...
+	DeactivateDormantUsersPeriod                          *int                      `url:"deactivate_dormant_users_period,omitempty" json:"deactivate_dormant_users_period,omitempty"`
+	DecompressArchiveFileTimeout                          *int                      `url:"decompress_archive_file_timeout,omitempty" json:"decompress_archive_file_timeout,omitempty"`
+	DefaultArtifactsExpireIn                              *string                   `url:"default_artifacts_expire_in,omitempty" json:"default_artifacts_expire_in,omitempty"`
+	DefaultBranchName                                     *string                   `url:"default_branch_name,omitempty" json:"default_branch_name,omitempty"`
+	DefaultBranchProtection                               *int                      `url:"default_branch_protection,omitempty" json:"default_branch_protection,omitempty"`
+	DefaultBranchProtectionDefaults                       *BranchProtectionDefaults `url:"default_branch_protection_defaults,omitempty" json:"default_branch_protection_defaults,omitempty"`
```
Comment (svanharmelen) [inline settings.go:548]: This field should be set to a `BranchProtectionDefaultsOptions` struct which contains the required fields as pointers (see other option structs).

**Review by svanharmelen — CHANGES_REQUESTED** [review]
Looks good, but one comment related to the additional struct.

File: types.go (hunk 1 above)
Comment (Jitsusama) [reply types.go]: This should be addressed now. Let me know if you would like it moved somewhere differently, I assumed you meant the bottom of the file, but I just recognized that you might have meant right below. I don't mind moving it to wherever you'd prefer.

File: settings.go (hunk 2 above)
Comment (Jitsusama) [reply settings.go]: This should be addressed now.

**Comment by Jitsusama** [comment]
@svanharmelen; mind taking another look at this, I think I have addressed all of your concerns and would love to cut another MR soon with the last batch of changes.

**Comment by timofurrer** [comment]
@Jitsusama while you are at it, do you mind adding `receptive_cluster_agents_enabled` that was recently added with https://gitlab.com/gitlab-org/gitlab/-/merge_requests/162514 ?

**Comment by Jitsusama** [comment]
> @Jitsusama while you are at it, do you mind adding `receptive_cluster_agents_enabled` that was recently added with https://gitlab.com/gitlab-org/gitlab/-/merge_requests/162514 ?

I'll add it to the next batch that covers the Rs @timofurrer!

**Review by svanharmelen — APPROVED** [review]
I updated the ordering (minor tweak), but looks good now 👍🏻  Thanks!

**Pull request merged**